### PR TITLE
add netlify config

### PR DESCRIPTION
**Pull request by moshloop**
/assign @detiber

### CODE REVIEW

File: netlify.toml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Netlify build instructions
+[build]
+    command = "./docs/book/build.sh"
+    publish = "docs/book/book"
+
+# Standard Netlify redirects
+[[redirects]]
+    from = "https://master--kubernetes-sigs-image-builder.netlify.com/*"
+    to = "https://master.image-builder.sigs.k8s.io/:splat"
```
Comment (detiber) [inline netlify.toml]: Should this reference just `image-builder.sigs.k8s.io` and `kubernetes-sigs-image-builder.netlify.com` for now? At least until we have another branch to publish from?

File: netlify.toml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+# HTTP-to-HTTPS rules
+[[redirects]]
+    from = "http://master.image-builder.sigs.k8s.io/*"
+    to = "https://master.image-builder.sigs.k8s.io/:splat"
+    status = 301
+    force = true
+
+[[redirects]]
+    from = "http://master--kubernetes-sigs-image-builder.netlify.com/*"
+    to = "http://master.image-builder.sigs.k8s.io/:splat"
```
Comment (detiber) [inline netlify.toml]: Should this reference just `image-builder.sigs.k8s.io` and `kubernetes-sigs-image-builder.netlify.com` for now? At least until we have another branch to publish from?

File: netlify.toml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+[[redirects]]
+    from = "https://master--kubernetes-sigs-image-builder.netlify.com/*"
+    to = "https://master.image-builder.sigs.k8s.io/:splat"
+    status = 301
+    force = true
+
+# HTTP-to-HTTPS rules
+[[redirects]]
+    from = "http://master.image-builder.sigs.k8s.io/*"
+    to = "https://master.image-builder.sigs.k8s.io/:splat"
```
Comment (detiber) [inline netlify.toml]: Should this reference just `image-builder.sigs.k8s.io` for now? At least until we have another branch to publish from?

**Comment by moshloop** [comment]
I think we want to release a v1 pretty soon and make sure the docs are stable because when v2 starts coming out it will be radically different.

**Comment by detiber** [comment]
> I think we want to release a v1 pretty soon and make sure the docs are stable because when v2 starts coming out it will be radically different.

Completely agree, but without another branch to point to as the "default" branch, `master` is the current default branch, so the referenced URLs probably will not work.

**Comment by moshloop** [comment]
@detiber done

**Comment by codenrhoden** [comment]
There was already a script for building the books at `hack/build-mdbook.sh`, and it downloaded a newer version of mdbook (`0.3.5` instead of `0.3.1`, though I see now that it echo's the wrong version). Apologies that I didn't get to this earlier. I'd like to see them de-dup'd.

**Pull request merged**